### PR TITLE
Update and rename python-app.yml to tests.yml

### PR DESCRIPTION
**Pull request by herjy**
An attempt at CI with the stack

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -10,23 +10,44 @@ on:
...
+        mamba install --quiet --yes --file requirements.txt
+        mamba install --quiet --yes --file requirements_dev.txt
+        mamba install -q -y stackvana=0 lsstdesc-weaklensingdeblending
+      
         python -m pip install --upgrade pip
         python -m pip install --upgrade pytest
         python -m pip install flake8 pytest pytest-cov coveralls
         if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-        if [ -f requirements.txt ]; then pip install -r test_requirements.txt; fi
+        if [ -f requirements.txt ]; then pip install -r test_requirements.txt;
```
Comment (sibirrer) [inline .github/workflows/tests.yml]: here you need to change the condition for the file name to test_requirements.txt as well

File: requirements.txt [hunk 2]
```diff
@@ -2,9 +2,9 @@
 numpy>=1.17
 astropy
 matplotlib
-lenstronomy
-galsim
 git+git://github.com/LSSTDESC/gcr-catalogs#egg=GCRCatalogs
+git+git://github.com/sibirrer/lenstronomy#egg=lenstronomy
```
Comment (sibirrer) [inline requirements.txt]: on the testing built failing: it looks like it wants to generate a conda environment and the cause of the failure is most likely that I have not registered a conda version of lenstronomy. Also note: there is a second requirement that does not have a direct conda version. If it is not necessary to have a conda built but instead go for the regular python built (you could e.g. look at the testing configuration of the skeleton package on the LSST-strong-lensing repo, or the lenstronomy source code)

Comment (sibirrer) [reply requirements.txt]: also, fyi, I just had troubles in lenstronomy as well to provide installations via git+git:// commands as it has some security guards in place recently.

**Review by sibirrer — CHANGES_REQUESTED** [review]
I gave two comments. Overall I am not sure the conda installation route allows or is looking for git packages directly. I simply don't have much experience. I am happy to provide a conda installation of lenstronomy, if this is really needed and the only way forward.

File: requirements.txt (hunk 2 above)
Comment (herjy) [reply requirements.txt]: Good to know. Hopefully, Heather's env will fix this. The issue is that we need an env here to have the stack running.

**Pull request closed without merging**